Revit_Engine/Query/ZJustifiedCurve.cs

```diff
@@ -49,7 +50,11 @@ public static ICurve ZJustifiedCurve(this IFramingElement element)
             else if (zJustification == "bottom")
                 return element.BottomCentreline();
             else if (zJustification == "center")
-                return element.GeometricalCentreline();
+            {
+                Point start = element.TopCentreline().IStartPoint().Translate(0.5 * (element.BottomCentreline().IStartPoint() - element.TopCentreline().IStartPoint()));
+                Point end = element.TopCentreline().IEndPoint().Translate(0.5 * (element.BottomCentreline().IEndPoint() - element.TopCentreline().IEndPoint()));
```

Comment:
I would consider avoiding excessive calls to `TopCentreline` and `BottomCentreline` if possible to avoid overhead. For example like this:
```suggestion
    Line top = element.TopCentreline();
    Line bottom = element.BottomCentreline();
    Point start = (top.IStartPoint() + bottom.IStartPoint()) * 0.5;
    Point end= (top.IEndPoint() + bottom.IEndPoint()) * 0.5;
```